public/pages/RollupDetails/components/GeneralInformation/GeneralInformation.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import React, { Component } from "react";
+import { EuiFlexGrid, EuiSpacer, EuiFlexItem, EuiText } from "@elastic/eui";
+import { ContentPanel, ContentPanelActions } from "../../../../components/ContentPanel";
+import { ModalConsumer } from "../../../../components/Modal";
+
+interface GeneralInformationProps {
+  rollupId: string;
+  description: string;
+  sourceIndex: string;
+  targetIndex: string;
+  scheduleText: string;
+  pageSize: number;
+  lastUpdated: string;
+  onEdit: () => void;
+}
+
+export default class GeneralInformation extends Component<GeneralInformationProps> {
+  constructor(props: GeneralInformationProps) {
+    super(props);
+  }
+
+  render() {
+    const { rollupId, description, onEdit, sourceIndex, targetIndex, scheduleText, pageSize, lastUpdated } = this.props;
+    return (
+      <ContentPanel
+        actions={
+          <ModalConsumer>
+            {() => (
+              <ContentPanelActions
+                actions={[
+                  {
+                    text: "Edit",
+                    buttonProps: {
+                      onClick: () => onEdit(),
+                    },
+                  },
+                ]}
+              />
+            )}
+          </ModalConsumer>
+        }
+        bodyStyles={{ padding: "initial" }}
+        title="General Information"
+        titleSize="m"
+      >
+        <div style={{ paddingLeft: "10px" }}>
```

Comment:
Define something like this above the return:
```js
const infoItems = [{ term: "Name", value: rollupId }, { term: "Source index", value: sourceIndex }, { term: "Target index", value: targetIndex }, { term: "Schedule", value: scheduleText }, { term: "Description", value: description || "-" }, { term: "Last updated", value: lastUpdated }]
```

And then instead of this long list of elements just do something like
```js
...
{infoItems.map((term, value) => (
  <EuiFlexItem>
    <EuiText size="xs>
      <dt>{term}</dt>
      <dd>{value}</dd>
    </EuiText>
  </EuiFlexItem>
))}
...
```

Reply:
Modified